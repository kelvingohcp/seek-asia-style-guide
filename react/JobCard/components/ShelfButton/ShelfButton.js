import React from 'react';
import PropTypes from 'prop-types';
import Text from '../../../Text/Text';
import Hidden from '../../../Hidden/Hidden';
import styles from './ShelfButton.less';
import classnames from 'classnames';
import ChevronIcon from '../../../ChevronIcon/ChevronIcon';
import Badge from '../../../Badge/Badge';
import { hasShelfLinks } from '../../jobCardHelper.js';
import { ShelfPropTypes } from '../ShelfSection/ShelfSectionPropTypes';

const ShelfButton = ({ desktopOnly, mobileOnly, isOpen, job, onClick }) => {
  return (
    <Hidden
      tablet={mobileOnly}
      mobile={desktopOnly}
      className={styles.root}>
      {job.featuredLabel ? (
        <span>
          <Badge color="new" label={job.featuredLabel} />
        </span>
      ) : (
        <Text
          whispering
          baseline={false}
          className={classnames(styles.shelfLinkText, job.appliedDate && styles.applyDate)}>
          {job.appliedDate || job.postingDuration}
        </Text>
      )}
      {hasShelfLinks(job.shelf && job.shelf.shelfLinks) && (
        <Hidden mobile className={styles.moreWrapper}>
          <Text whispering className={styles.separatorDot} baseline={false}>â€¢</Text>
          <Text
            whispering
            className={classnames(styles.shelfLinkText, styles.shelfToggle)}
            baseline={false}
            onClick={onClick}>
            {isOpen ? 'less' : 'more'}{' '}
            <ChevronIcon
              direction={isOpen ? 'up' : 'down'}
              svgClassName={styles.shelfToggleIcon}
            />
          </Text>
        </Hidden>
      )}
    </Hidden>
  );
};

ShelfButton.propTypes = {
  desktopOnly: PropTypes.bool,
  mobileOnly: PropTypes.bool,
  job: PropTypes.shape({
    featuredLabel: PropTypes.string,
    appliedDate: PropTypes.string,
    postingDuration: PropTypes.string,
    shelf: ShelfPropTypes
  }),
  onClick: PropTypes.func,
  isOpen: PropTypes.bool
};
export default ShelfButton;
