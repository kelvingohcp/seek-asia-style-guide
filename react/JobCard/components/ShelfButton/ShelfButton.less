@import (reference) '~seek-asia-style-guide/theme';
@import (reference) '../../JobCard.less';

.shelfToggleIcon {
  width: @grid-base * 2;
  height: @grid-base * 2;
  margin-left: @grid-base;
  @supports (display: grid) {
    margin-left: unset;
  }
}

.separatorDot {
  color: @saGrey4;
  display: inline;
  padding: 0 @grid-base * 2;
  @supports (display: grid) {
    padding: 0;
    margin-left: @grid-base;
  }
}

.shelfToggle {
  .link(@saGrey2);
  align-items: center;
  display: flex;
  height: unset;
  &:focus {
    outline: none;
  }
}

.shelfLinkText {
  padding-bottom: 0;
  color: @saGrey2;
  white-space: nowrap;
}

.applyDate {
  color: @saGrey1;
}
