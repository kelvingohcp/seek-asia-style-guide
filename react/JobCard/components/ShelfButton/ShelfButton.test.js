import React from 'react';
import { shallow } from 'enzyme';

import ShelfButton from './ShelfButton';

const defaultProps = {
  desktopOnly: false,
  mobileOnly: false,
  isOpen: false,
  job: {
    postingDuration: '1 hour ago'
  },
  onClick: () => {}
};

const shelf = {
  shelfLinks: [
    {
      label: 'Job function',
      items: [
        {
          name: 'Accounting',
          link: '/jobCard',
          title: 'Limit result to Accounting',
          children: [
            {
              name: 'Accountant',
              link: '/jobCard',
              title: 'Limit result to Accountant in Accounting'
            },
            {
              name: 'Audit',
              link: '/jobCard',
              title: 'Limit result to Audit in Accounting'
            },
            {
              name: 'Others',
              link: '/jobCard',
              title: 'Limit result to Others in Accounting'
            }
          ]
        },
        {
          name: 'Admin & HR',
          link: '/jobCard',
          title: 'Limit result to Admin & HR',
          children: [
            {
              name: 'Receptionist',
              link: '/jobCard',
              title: 'Limit result to Receptionist in Admin & HR'
            }
          ]
        }
      ]
    },
    {
      label: 'Industry',
      items: [{
        name: 'Accounting / Audit / Tax Services',
        link: '/jobCard',
        title: 'Limit result to Accounting / Audit / Tax Services'
      }]
    }
  ]
};

describe('Shelf Button', () => {
  it('should render default shelf button', () => {
    const wrapper = shallow((<ShelfButton {...defaultProps} />));
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Shelf Button', () => {
  it('should render shelf button with featured label', () => {
    const wrapper = shallow((<ShelfButton {...defaultProps} featuredLabel="Featured" />));
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Shelf Button', () => {
  it('should render shelf button with applied date', () => {
    const wrapper = shallow((<ShelfButton {...defaultProps} appliedDate="Applied on 20-Jul-2018" />));
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Shelf Button', () => {
  it('should render shelf button with shelf', () => {
    const wrapper = shallow((<ShelfButton {...defaultProps} job={{ ...defaultProps.job, shelf }} />));
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Shelf Button', () => {
  it('should render shelf button with shelf and openned style', () => {
    const wrapper = shallow((<ShelfButton {...defaultProps} job={{ ...defaultProps.job, shelf }} isOpen />));
    expect(wrapper).toMatchSnapshot();
  });
});
