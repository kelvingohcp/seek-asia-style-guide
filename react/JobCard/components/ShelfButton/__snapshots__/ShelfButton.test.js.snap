// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shelf Button should render default shelf button 1`] = `
<Hidden
  aboveMobile={false}
  className="root"
  component="span"
  desktop={false}
  mobile={false}
  print={false}
  screen={false}
>
  <Text
    baseline={false}
    className="shelfLinkText"
    whispering={true}
  >
    1 hour ago
  </Text>
</Hidden>
`;

exports[`Shelf Button should render shelf button with applied date 1`] = `
<Hidden
  aboveMobile={false}
  className="root"
  component="span"
  desktop={false}
  mobile={false}
  print={false}
  screen={false}
>
  <Text
    baseline={false}
    className="shelfLinkText"
    whispering={true}
  >
    1 hour ago
  </Text>
</Hidden>
`;

exports[`Shelf Button should render shelf button with featured label 1`] = `
<Hidden
  aboveMobile={false}
  className="root"
  component="span"
  desktop={false}
  mobile={false}
  print={false}
  screen={false}
>
  <Text
    baseline={false}
    className="shelfLinkText"
    whispering={true}
  >
    1 hour ago
  </Text>
</Hidden>
`;

exports[`Shelf Button should render shelf button with shelf 1`] = `
<Hidden
  aboveMobile={false}
  className="root"
  component="span"
  desktop={false}
  mobile={false}
  print={false}
  screen={false}
>
  <Text
    baseline={false}
    className="shelfLinkText"
    whispering={true}
  >
    1 hour ago
  </Text>
  <Hidden
    aboveMobile={false}
    className="moreWrapper"
    component="span"
    desktop={false}
    mobile={true}
    print={false}
    screen={false}
  >
    <Text
      baseline={false}
      className="separatorDot"
      whispering={true}
    >
      •
    </Text>
    <Text
      baseline={false}
      className="shelfLinkText shelfToggle"
      onClick={[Function]}
      whispering={true}
    >
      more
       
      <ChevronIcon
        className=""
        direction="down"
        svgClassName="shelfToggleIcon"
      />
    </Text>
  </Hidden>
</Hidden>
`;

exports[`Shelf Button should render shelf button with shelf and openned style 1`] = `
<Hidden
  aboveMobile={false}
  className="root"
  component="span"
  desktop={false}
  mobile={false}
  print={false}
  screen={false}
>
  <Text
    baseline={false}
    className="shelfLinkText"
    whispering={true}
  >
    1 hour ago
  </Text>
  <Hidden
    aboveMobile={false}
    className="moreWrapper"
    component="span"
    desktop={false}
    mobile={true}
    print={false}
    screen={false}
  >
    <Text
      baseline={false}
      className="separatorDot"
      whispering={true}
    >
      •
    </Text>
    <Text
      baseline={false}
      className="shelfLinkText shelfToggle"
      onClick={[Function]}
      whispering={true}
    >
      less
       
      <ChevronIcon
        className=""
        direction="up"
        svgClassName="shelfToggleIcon"
      />
    </Text>
  </Hidden>
</Hidden>
`;
